llvm/include/llvm/Analysis/TargetTransformInfoImpl.h

```diff
@@ -895,6 +897,13 @@ class TargetTransformInfoImplBase {
 
   unsigned getMaxNumArgs() const { return UINT_MAX; }
 
+  std::optional<KnownBits>
+  computeKnownBitsAddrSpaceCast(unsigned DestAS, unsigned SrcAS,
+                                const APInt &DemandedElts, KnownBits &Known,
+                                const SimplifyQuery &Q) const {
+    return std::nullopt;
```

Comment:
The default implementation might consider checking the alignment for integral address spaces